source/common/config/tls_context_json.cc

```diff
@@ -33,10 +33,9 @@ void TlsContextJson::translateUpstreamTlsContext(
 
 void TlsContextJson::translateCommonTlsContext(
     const Json::Object& json_tls_context, envoy::api::v2::CommonTlsContext& common_tls_context) {
-  const std::vector<std::string> alpn_protocols =
-      StringUtil::split(json_tls_context.getString("alpn_protocols", ""), ",");
-  for (const auto& alpn_protocol : alpn_protocols) {
-    common_tls_context.add_alpn_protocols(alpn_protocol);
+  for (auto alpn_protocol :
+       StringUtil::splitToken(json_tls_context.getString("alpn_protocols", ""), ",")) {
```

Comment:
I have a guess as to the ASAN problem.  Can you make a std::string temp for json_tls_context.getString("alpn_protocols", "") outside the for-loop?

The one thing you need to be sure of with string_view is that the backing-store for the chars outlives the uses of the string_view.

Reply:
I've done that yesterday, but I think it might be something else. Let me kick the CI off again. Things in my machine are very slow.